### PR TITLE
Add helper methods for timeseries support

### PR DESCRIPTION
**Pull request by justinGilmer**
This adds in helper functions to detect equilibration in timeseries
data, as well as trimming away the timeseries data that is not
equilibrated.

Unit tests are included to verify behavior.

### CODE REVIEW

**Comment by justinGilmer** [comment]
@chrisjonesBSU can you take a look at this and see if we can leverage utilities like these in your PR? This should provide a defined method we can all use in the study to then sample data without much code duplication for different properties.

File: project/src/analysis/equlibration.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    else:
+        return False
+
+
+def trim_non_equilibrated(
+    a_t: npt.ArrayLike, threshold: float = 0.75, nskip: int = 1
+) -> List:
+    """Prune timeseries array to just the production data.
+
+    Refer to equilibration.is_equilibrated for addtional information.#!/usr/bin/env python
```
Comment (jennyfothergill) [inline project/src/analysis/equlibration.py:55]:
```suggestion
    Refer to equilibration.is_equilibrated for addtional information.
```

File: project/tests/test_equilbration.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+    def test_incorrect_threshold(self):
+        data = testsystems.correlated_timeseries_example(
+            N=1000, tau=200, seed=432
+        )
+        with pytest.raises(ValueError, match=r"Passed \'threshold\' value"):
+            is_equilibrated(data, threshold=2.0)
+
+        with pytest.raises(ValueError, match=r"Passed \'threshold\' value"):
+            is_equilibrated(data, threshold=-2.0)
```
Comment (jennyfothergill) [inline project/tests/test_equilbration.py:30]: is one of these supposed to be negative?

**Review by jennyfothergill — APPROVED** [review]

File: project/src/analysis/equlibration.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    threshold : float, optional, default=0.8
+        Fraction of data expected to be equilibrated.
+    nskip : int, optional, default=1
+        Since the statistical inefficiency is computed for every time origin
+        in a call to timeseries.detectEquilibration, for larger datasets
+        (> few hundred), increasing nskip might speed this up, while
+        discarding more data.
+
+    """
+    if not is_equilibrated(a_t, threshold=threshold, nskip=nskip):
```
Comment (chrisjonesBSU) [inline project/src/analysis/equlibration.py:78]: When calling the `is_equilibrated` function, if it's True, could it also return `t0` and `g`?  That way we don't have to run `timeseries.detectEquilibration` a second time in `trim_non_equilibrated`

File: project/tests/test_equilbration.py (hunk 2 above)
Comment (justinGilmer) [reply project/tests/test_equilbration.py:30]: Yeah I just wanted to make sure my validation check for `threshold` was behaving as expected. Only allow floats between 0.0-1.0

File: project/src/analysis/equlibration.py (hunk 3 above)
Comment (justinGilmer) [reply project/src/analysis/equlibration.py:78]: I haven’t tried that in python, but would that be an issue for the false case? 

Or I guess we can just return None for those values. Does `None` for the non-existent t0 and g seem reasonable in the false case?

File: project/src/analysis/equlibration.py (hunk 1 above)
Comment (justinGilmer) [reply project/src/analysis/equlibration.py:55]: Good catch, my ide autocomplete thought it was being helpful with the tab completion 😂

File: project/src/analysis/equlibration.py (hunk 3 above)
Comment (jennyfothergill) [reply project/src/analysis/equlibration.py:78]: yeah, `None` seems reasonable-- in this case we would use it something like this?
```
eq, t0, g = is_equilibrated
if eq:
    do something
```

Comment (justinGilmer) [reply project/src/analysis/equlibration.py]: Ok cool. I just pushed those changes!

**Pull request merged**